### PR TITLE
Fix the Maven build

### PR DESCRIPTION
**Pull request by fbricon**
The uberfire runtime jars are now defined in a separate bundle (same way m2e builds).
This has some pros and cons:

Pros:
- simpler dependency management
- no more ant build
- automatic integration with m2e (requires the m2eclipse-tycho plugin)

Cons:
- requires a 2-step build: the runtime needs to be built/installed before the consumer plugins.
  This is a limitation from Tycho. See https://wiki.eclipse.org/Tycho/How_Tos/Dependency_on_pom-first_artifacts#It_is_not_possible_to_mix_pom-first_and_manifest-first_projects_in_the_same_reactor_build.

Signed-off-by: Fred Bricon fbricon@gmail.com

### CODE REVIEW

**Comment by fbricon** [comment]
I also fixed the feature and update site which were broken

File: eclipse-plugins/org.uberfire.eclipse.runtime/pom.xml [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+						<Bundle-RequiredExecutionEnvironment>JavaSE-1.8</Bundle-RequiredExecutionEnvironment>
+						<Bundle-Name>%Bundle-Name</Bundle-Name>
+						<Bundle-Vendor>%Bundle-Vendor</Bundle-Vendor>
+						<Bundle-ClassPath>{maven-dependencies}</Bundle-ClassPath>
+						<Eclipse-BundleShape>dir</Eclipse-BundleShape>
+						<archive>
+							<addMavenDescriptor>false</addMavenDescriptor>
+						</archive>
+						<Embed-Dependency>
+							*;scope=compile|runtime;artifactId=!slf4j-api|javax.inject|
```
Comment (fbricon) [inline eclipse-plugins/org.uberfire.eclipse.runtime/pom.xml:81]: there's most certainly some cleanup to do here, you probably don't want to embed all the transitive dependencies. I'll leave that to the future maintainer.

File: eclipse-plugins/org.uberfire.eclipse.runtime/pom.xml [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+						</Embed-Dependency>
+						<Embed-Directory>lib</Embed-Directory>
+						<_exportcontents>
```
Comment (fbricon) [inline eclipse-plugins/org.uberfire.eclipse.runtime/pom.xml:84]: these are the packages currently used by the browser plugin. There's probably some more fine tuning to do here.

File: eclipse-plugins/org.uberfire.eclipse.runtime/pom.xml [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+							org.drools.*;provider=uberfire;mandatory:=provider,
+							org.uberfire.*;provider=uberfire;mandatory:=provider,
+							org.guvnor.*;provider=uberfire;mandatory:=provider,
+						</_exportcontents>
+						<Import-Package>
+							!*
```
Comment (fbricon) [inline eclipse-plugins/org.uberfire.eclipse.runtime/pom.xml:90]: !\* means no packages are imported. You'll get some errors when building the update site if this statement is removed

File: eclipse-plugins/org.uberfire.eclipse.runtime/src/main/resources/about.html [hunk 4]
```diff
@@ -0,0 +1,117 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
```
Comment (fbricon) [inline eclipse-plugins/org.uberfire.eclipse.runtime/src/main/resources/about.html:1]: this was copied from m2e. might not apply here

File: eclipse-plugins/org.uberfire.eclipse.runtime/src/main/resources/about_files/LICENSE-2.0.txt [hunk 5]
```diff
@@ -0,0 +1,202 @@
+
```
Comment (fbricon) [inline eclipse-plugins/org.uberfire.eclipse.runtime/src/main/resources/about_files/LICENSE-2.0.txt:1]: this was copied from m2e. might not apply here

**Comment by bbrodt** [comment]
Thanks a bunch Fred! I'll test this out today. Hopefully I can take it from here ;)

**Pull request merged**